app/models/ingredient.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+class Ingredient < ApplicationRecord
+  DOES_NOT_MATCH = "doesn't match an ingredient that exists in Skyrim"
+
+  belongs_to :game
+  belongs_to :canonical_ingredient,
+             class_name: 'Canonical::Ingredient',
+             optional: true,
+             inverse_of: :ingredients
+
+  has_many :ingredients_alchemical_properties, dependent: :destroy, inverse_of: :ingredient
+  has_many :alchemical_properties,
+           -> { select 'alchemical_properties.*, ingredients_alchemical_properties.priority' },
+           through: :ingredients_alchemical_properties
+
+  validates :name, presence: true
+  validate :ensure_match_exists
+
+  before_validation :set_canonical_ingredient
+
+  def canonical_ingredients
+    return Canonical::Ingredient.where(id: canonical_ingredient.id) if canonical_ingredient.present?
```

Comment:
Other non-canonical models would do it like this:
```ruby
return Array.wrap(canonical_ingredient) if canonical_ingredient.present?
```
Why do we need an `ActiveRecord::Relation` here?

Reply:
Answer: because we call `canonical_ingredients.ids` in the `IngredientsAlchemicalProperty` class and that method is only available on an `ActiveRecord::Relation`.